edit-and-sync-features-with-feature-service/src/main/java/com/esri/arcgismaps/sample/editandsyncfeatureswithfeatureservice/components/MapViewModel.kt

```diff
@@ -0,0 +1,465 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.editandsyncfeatureswithfeatureservice.components
+
+import android.app.Application
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.data.Feature
+import com.arcgismaps.data.Geodatabase
+import com.arcgismaps.geometry.Envelope
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.Polygon
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.ScreenCoordinate
+import com.arcgismaps.mapping.view.SingleTapConfirmedEvent
+import com.arcgismaps.tasks.geodatabase.GenerateGeodatabaseJob
+import com.arcgismaps.tasks.geodatabase.GeodatabaseSyncTask
+import com.arcgismaps.tasks.geodatabase.SyncDirection
+import com.arcgismaps.tasks.geodatabase.SyncGeodatabaseJob
+import com.arcgismaps.tasks.geodatabase.SyncGeodatabaseParameters
+import com.arcgismaps.tasks.geodatabase.SyncLayerOption
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.editandsyncfeatureswithfeatureservice.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.io.File
+
+class MapViewModel(private val application: Application) : AndroidViewModel(application) {
+
+    // Create a red boundary line showing the filtered map extents for the features
+    private val boundarySymbol = SimpleLineSymbol(
+        style = SimpleLineSymbolStyle.Solid,
+        color = Color.red,
+        width = 5f
+    )
+
+    // Create graphic overlay to add graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    // Create a ViewModel to handle dialog interactions.
+    val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
```

Comment:
Could be simplified: 

```suggestion
    val messageDialogVM = MessageDialogViewModel()
```